lib/index.js

```diff
@@ -38,6 +42,35 @@ function initialize(config) {
   });
 }
 
+function filterPreviousUploads(uploadListPath, objects) {
+  var uploadedFileList;
+  if (uploadListPath) {
+    uploadedFileList = fs.readFileSync(uploadListPath, {encoding: 'utf8'});
+    uploadedFileList = uploadedFileList.split('\n');
+    if (!uploadedFileList[uploadedFileList.length-1]) uploadedFileList.pop();
+
+    uploadedFileList = _.map(uploadedFileList, function(anUploadedFile) {
+      return JSON.parse(anUploadedFile);
+    });
+
+    uploadedFilesById = _.groupBy(uploadedFileList, 'objectId');
+  }
+
+  objects = _.filter(objects, function(aFileObject) {
+    if (!aFileObject.fileName) return false; //null fileName causes error in indexOf later
```

Comment:
yeah I agree,  we can add to the chain in `getAllFileObjects`.  The benefit in there is that it's clear what's happening, like after all the fetching, we are adding a filtering / checking for validation.  I think that would make the intention more clear, and it's technically a part of the intention of "getAllFilterObjects"  What do you think?